pkg/handler/collector/deps_dev/deps_dev.go

```diff
@@ -132,6 +143,102 @@ func (d *depsCollector) populatePurls(ctx context.Context, docChannel chan<- *pr
 	return nil
 }
 
+// getAllDependencies gets all the dependencies for the purls provided in a concurrent manner.
+func (d *depsCollector) getAllDependencies(ctx context.Context, purls []datasource.Source) error {
+	// channels to signal when the project and version info have been fetched
+	projectDone := make(chan bool)
+	versionDone := make(chan bool)
+
+	// channels to send the inputs to the goroutines
+	projectChan := make(chan *pb.ProjectKey)
+	versionChan := make(chan *pb.VersionKey)
+	logger := logging.FromContext(ctx)
+
+	// these goroutines will be used to version info concurrently
+	go func() {
+		// this go routine has to be before the next go routine as it will be pushing into the project channel
+		// for each version that is fetched from the version channel it will check if the project has to be fetched
+		d.versions = d.getVersions(ctx, versionChan, projectChan) // the results are the stored in the versions map
+		versionDone <- true
+	}()
+
+	// these goroutines will be used to fetch the projects concurrently
+	go func() {
+		// this sets up the goroutine to fetch the projects concurrently for each input
+		d.projectInfoMap = d.getProjects(ctx, projectChan) // the results are the stored in the projectInfoMap map
+		// posts to the projectDone channel to signal that all projects have been fetched
+		projectDone <- true
+	}()
+
+	// TODO: Concurrently fetch the dependencies for each purl
```

Comment:
qq: did we want to keep the issue open to complete this TODO as part of it?

Reply:
- I want to get this PR merged and would like to take this as a separate issue. 

- I plan to get some actual unit tests with mocks and would like to target it with that. This will be a subsequent PR.

- We can compare what it is now and later decide whether it makes sense to do it or not. 

- The only way to answer that is by running through the blocking profiler https://github.com/DataDog/go-profiler-notes/blob/main/block.md, which will help us identify the blocked network calls. 

Thank you for the quick turnaround. I appreciate that.